src/transformStateWithClones.js

```diff
@@ -7,7 +7,39 @@
  * @return {Object[]}
  */
 function transformStateWithClones(state, actions) {
-  // write code here
+  const transformedStates = [];
+  const stateCopy = { ...state };
+
+  for (const action of actions) {
+    switch (action.type) {
+      case 'clear':
+        Object.keys(stateCopy)
+          .forEach(key => delete stateCopy[key]);
+        transformedStates.push({ ...stateCopy });
```

Comment:
move `transformedStates.push({ ...stateCopy });` after switch case instead of using it for each case